src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
Actually, a thought just came to mind: could we break a swap down into two `move_node_to_parent` calls (plus some extra logic and minor tweaks)?

Still on mobile so I can't draw it out but I think it might work fairly well. If it does, then we might be able to logic out the behavior a bit more easily.